src/tests/java/htsjdk/samtools/fastq/FastqRecordTest.java

```diff
@@ -0,0 +1,136 @@
+        Assert.assertNotEquals("fred", fastqRecord);
+        Assert.assertEquals(fastqRecord.length(), seqLine.length());
+    }
+
+    @Test
+    public void testBasicEmptyHeaderPrefix() {
+        final String seqHeaderPrefix = "";
+        final String seqLine = "ACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGT";
+        final String qualHeaderPrefix = "";
+        final String qualLine = ";<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\\]^_`abcdefghijklmnopqrstuvwxyz{|}~";
+        final FastqRecord fastqRecord = new FastqRecord(seqHeaderPrefix, seqLine, qualHeaderPrefix, qualLine);
+        Assert.assertNull(fastqRecord.getReadHeader());
+        Assert.assertNull(fastqRecord.getBaseQualityHeader());
+    }
+
+    @Test
+    public void testCopy() {
+        final String seqHeaderPrefix = "FAKE0003 Original version has Solexa scores from 62 to -5 inclusive (in that order)";
+        final String seqLine = "ACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTACGT";
+        final String qualHeaderPrefix = "";
+        final String qualLine = ";<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\\]^_`abcdefghijklmnopqrstuvwxyz{|}~";
+        final FastqRecord fastqRecord = new FastqRecord(seqHeaderPrefix, seqLine, qualHeaderPrefix, qualLine);
+        final FastqRecord fastqRecordCopy = new FastqRecord(fastqRecord);
+
+        Assert.assertEquals(fastqRecord, fastqRecordCopy);
+        Assert.assertNotSame(fastqRecord, fastqRecordCopy);
+        Assert.assertSame(fastqRecord.getReadString(), fastqRecordCopy.getReadString());
+        Assert.assertSame(fastqRecord.getBaseQualityString(), fastqRecordCopy.getBaseQualityString());
+        Assert.assertSame(fastqRecord.getBaseQualityHeader(), fastqRecordCopy.getBaseQualityHeader());
+    }
+
+    @Test
+    public void testCopyWithNullSeq() {
+        final String seqHeaderPrefix = "header";
+        final String seqLine = null;
+        final String qualHeaderPrefix = "";
+        final String qualLine = ";<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\\]^_`abcdefghijklmnopqrstuvwxyz{|}~";
+        final FastqRecord fastqRecord = new FastqRecord(seqHeaderPrefix, seqLine, qualHeaderPrefix, qualLine);
+        final FastqRecord fastqRecordCopy = new FastqRecord(fastqRecord);
+
+        Assert.assertEquals(fastqRecord, fastqRecordCopy);
+        Assert.assertNotSame(fastqRecord, fastqRecordCopy);
+        Assert.assertSame(fastqRecord.getReadString(), fastqRecordCopy.getReadString());
+        Assert.assertSame(fastqRecord.getBaseQualityString(), fastqRecordCopy.getBaseQualityString());
+        Assert.assertSame(fastqRecord.getBaseQualityHeader(), fastqRecordCopy.getBaseQualityHeader());
+    }
+
+    @Test
+    public void testEqualsWithNullSeq() {
+        final String seqHeaderPrefix = "header";
+        final String seqLine = "GATTACA";
+        final String qualHeaderPrefix = "";
+        final String qualLine = ";<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\\]^_`abcdefghijklmnopqrstuvwxyz{|}~";
```

Comment:
Various sites (ex. wikipedia, MAQ's page, and even here [1]) all say that the seq and qual fields should be the same length, so I would prefer a change in semantics.  `FastqReader` also enforces this prior to creating a `FastqRecord`.

[1] http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2847217/